nbfetch/handlers.py

```diff
@@ -304,43 +323,40 @@ def login(self):
         # check for oauth
         authfile = os.path.expanduser("~/.hs_auth")
         try:
-            with open(authfile, 'rb') as f:
+            with open(authfile, "rb") as f:
                 token, cid = pickle.load(f)
-            auth = HydroShareAuthOAuth2(cid, '', token=token)
-            self.log.info('auth=%s' % str(auth))
+            auth = HydroShareAuthOAuth2(cid, "", token=token)
+            self.log.info("auth=%s" % str(auth))
 
             hs = self.check_auth(auth)
-            self.log.info('hs=%s' % str(hs))
+            self.log.info("hs=%s" % str(hs))
 
             if hs is None:
-                message = url_escape("Oauth Login Failed.  Login with username and password or logout from JupyterHub and reauthenticate with Hydroshare.")
+                message = url_escape(
+                    "Oauth Login Failed.  Login with username and password or logout from JupyterHub and reauthenticate with Hydroshare."
+                )
         except:
-            message = ''
+            message = ""
 
         if hs is None:
             # If oauth fails, we can log in using
             # user name and password.  These are saved in
             # files in the home directory.
-            pwfile = os.path.expanduser("~/.hs_pass")
-            userfile = os.path.expanduser("~/.hs_user")
-
+            username, password = _get_user_authentication()
             try:
-                with open(userfile) as f:
-                    username = f.read().strip()
-                with open(pwfile) as f:
-                    password = f.read().strip()
                 auth = HydroShareAuthBasic(username=username, password=password)
                 hs = self.check_auth(auth)
                 if hs is None:
                     message = url_escape("Login Failed. Please Try again")
             except:
-                message = url_escape("You need to provide login credentials to access HydroShare Resources.")
+                message = url_escape(
+                    "You need to provide login credentials to access HydroShare Resources."
+                )
 
         if hs is None:
             _next = url_escape(url_escape(self.request.uri))
-            upath = urljoin(self.request.uri, 'hslogin')
-            self.redirect('%s?error=%s&next=%s' % (upath, message, _next))
-
+            upath = urljoin(self.request.uri, "hslogin")
+            self.redirect("%s?error=%s&next=%s" % (upath, message, _next))
```

Comment:
It's not a huge deal, but I want us to migrate to f-strings instead of %-formatting. They are easier to 
 understand as strings get longer and we tend to use them more often than not. This will prevent us from using multiple string formatting approaches in the future.

Reply:
Totally agree, this is not my doing. From the old code base. I think there is a lot of minor fix-ups like this ridden around this code-base that we should fix up.